tests/api.c

```diff
@@ -7645,6 +7645,538 @@ static int test_wolfSSL_CTX_add_session(void)
 
     return res;
 }
+#if defined(HAVE_IO_TESTS_DEPENDENCIES) && defined(HAVE_EXT_CACHE) && \
+    !defined(SINGLE_THREADED) && defined(WOLFSSL_TLS13) && \
+    !defined(NO_SESSION_CACHE) && defined(OPENSSL_EXTRA) && \
+     defined(SESSION_CERTS) && defined(HAVE_SESSION_TICKET) && \
+    !defined(TITAN_SESSION_CACHE) && \
+    !defined(HUGE_SESSION_CACHE) && \
+    !defined(BIG_SESSION_CACHE) && \
+    !defined(MEDIUM_SESSION_CACHE)
+
+/* Sessions to restore/store */
+static WOLFSSL_SESSION* server_first_session;
+static WOLFSSL_SESSION* client_first_session;
+static WOLFSSL_CTX*     server_current_ctx;
+static int new_session_called    = 0;
+static int remove_session_called = 0;
+static int get_session_called    = 0;
+
+/* Test default, SESSIONS_PER_ROW*SESSION_ROWS = 3*11, see ssl.c */
+#define SESSION_CACHE_SIZE 33
+
+typedef struct {
+    const byte* key;  /* key, altSessionID, session ID, NULL if empty */
+    WOLFSSL_SESSION* value;
+} hashTable_entry;
+
+typedef struct {
+    hashTable_entry entries[SESSION_CACHE_SIZE];  /* hash slots */
+    size_t capacity;                     /* size of entries */
+    size_t length;                       /* number of items in the hash table */
+    wolfSSL_Mutex htLock;                /* lock */
+}hashTable;
+
+static hashTable server_sessionCache;
+
+static int new_sessionCb(WOLFSSL *ssl, WOLFSSL_SESSION *sess)
```

Comment:
Please namespace these functions a little with a common unique prefix. Unfortunately C doesn't provide us a way to namespace and this is global to the entire `api.c`.

Reply:
@julek-wolfssl, Good suggestion. I initially wanted to avoid cluttered code with long function names for clarity but I could use your short prefix.